src/main/java/mekanism/common/tile/prefab/TileEntityStructuralMultiblock.java

```diff
@@ -169,18 +180,23 @@ public void onNeighborChange(Block block, BlockPos neighborPos) {
 
     @Override
     public InteractionResult onRightClick(Player player) {
-        if (!isRemote()) {
-            for (Structure s : structures.values()) {
-                if (s.getController() != null && !getMultiblockData(s).isFormed()) {
-                    FormationResult result = s.runUpdate(this);
+        if (isRemote()) {
+            return InteractionResult.PASS;
+        }
+        InteractionResult interactionResult = InteractionResult.PASS;
+        for (Structure structure : structures.values()) {
+            if (interactionResult == InteractionResult.PASS && structure.getController() != null) {
+                MultiblockData multiblock = getMultiblockData(structure);
+                if (multiblock == null || !multiblock.isFormed()) {
+                    FormationResult result = structure.runUpdate(this);
                     if (!result.isFormed() && result.getResultText() != null) {
                         player.sendSystemMessage(result.getResultText());
-                        return InteractionResult.sidedSuccess(isRemote());
+                        interactionResult = InteractionResult.sidedSuccess(isRemote());
```

Comment:
I don't see a reason to assign it to a local?

Reply:
Similar reason to the one for gui interaction, but I will apply that comment to both places where it just does an early return.